packages/@glimmer/runtime/lib/helpers/get.ts

```diff
@@ -0,0 +1,131 @@
+      name: "Sandi Metz",
+      language: "Ruby"
+    }
+  }
+  ```
+
+  ```handlebars
+  {{this.developer.name}}
+  {{get this.developer "name"}}
+  ```
+
+  If there were several facts about a person, the `{{get}}` helper can dynamically
+  pick one:
+
+  ```app/templates/application.hbs
+  <DeveloperDetail @factName="language" />
+  ```
+
+  ```handlebars
+  {{get this.developer @factName}}
+  ```
+
+  For a more complex example, this template would allow the user to switch
+  between showing the user's height and weight with a click:
+
+  ```app/components/developer-detail.js
+  import Component from '@glimmer/component';
+  import { tracked } from '@glimmer/tracking';
+
+  export default class extends Component {
+    @tracked developer = {
+      name: "Sandi Metz",
+      language: "Ruby"
+    }
+
+    @tracked currentFact = 'name'
+
+    @action
+    showFact(fact) {
+      this.currentFact = fact;
+    }
+  }
+  ```
+
+  ```app/components/developer-detail.js
+  {{get this.developer this.currentFact}}
+
+  <button {{on 'click' (fn this.showFact "name")}}>Show name</button>
+  <button {{on 'click' (fn this.showFact "language")}}>Show language</button>
+  ```
+
+  The `{{get}}` helper can also respect mutable values itself. For example:
+
+  ```app/components/developer-detail.js
+  <Input @value={{mut (get this.person this.currentFact)}} />
+
+  <button {{on 'click' (fn this.showFact "name")}}>Show name</button>
+  <button {{on 'click' (fn this.showFact "language")}}>Show language</button>
+  ```
+
+  Would allow the user to swap what fact is being displayed, and also edit
+  that fact via a two-way mutable binding.
+
+  @public
+  @method get
+ */
+export default internalHelper((args: VMArguments) => {
+  let sourceRef = args.positional.at(0);
+  let pathRef = args.positional.at(1);
+
+  if (isConstRef(pathRef)) {
+    // Since the path is constant, we can create a normal chain of property
+    // references. The source reference will update like normal, and all of the
+    // child references will update accordingly.
+    let path = valueForRef(pathRef);
+
+    if (path === undefined || path === null || path === '') {
+      return NULL_REFERENCE;
+    } else if (typeof path === 'string' && path.indexOf('.') > -1) {
+      return childRefFromParts(sourceRef, path.split('.'));
+    } else {
+      return childRefFor(sourceRef, String(path));
+    }
+  } else {
+    return createComputeRef(
+      () => {
+        let source = valueForRef(sourceRef);
+
+        if (isObject(source)) {
+          return getPath(source, String(valueForRef(pathRef)));
+        }
+      },
+      (value) => {
+        let source = valueForRef(sourceRef);
+
+        if (isObject(source)) {
+          return setPath(source, String(valueForRef(pathRef)), value);
+        }
+      },
+      'get'
+    );
+  }
+});
+
+function isObject(obj: unknown): obj is object {
+  return typeof obj === 'function' || (typeof obj === 'object' && obj !== null);
+}
```

Comment:
Don't we have a util function for this somewhere?

Reply:
we have one that is `isObject` in `@glimmer/util`, but it does not check if the value is a `function` 🙃 I was planning on doing a sweep to deduplicate it soon (and also confirm with @krisselden that deduplicating it wouldn't cause any perf issues)

Reply:
Opened up an issue for this: https://github.com/glimmerjs/glimmer-vm/issues/1246